Update stanchion shared code

kellymclaughlin opened the pull request:
Update the shared code ugliness.

kellymclaughlin committed:
Updated shared code from latest changes in riak_moss.

kellymclaughlin committed:
Update some module names from cut and paste.

reiddraper commented:
Something appears to be up with `bucket_empty`. If I recompile both moss and stanchion with `-compile(export_all).` and run:

``` erlang
{ok, Pid} = riak_moss_utils:riak_connection().
riak_moss_utils:bucket_empty(<<"test">>, Pid).
```

moss returns true and stanchion returns false (the bucket is not empty). This got me looking at the code some
more, and it seems like `stanchion_utils` has fallen behind.

reiddraper committed:
Fix stanchion_utils inconsistencies w/ riak_moss_utils

* use start_link/3
* update bucket name function

reiddraper committed:
Don't prune manifests

We don't have access (b/c we haven't copied it over)
to riak_cs_gc, so we don't want to prune
manifests

reiddraper commented:
+1
